inc/db.function.php

```diff
@@ -1832,77 +1376,29 @@ function getEntitiesRestrictRequest($separator = "AND", $table = "", $field = ""
  *                            (false by default)
  *
  * @return array of criteria
+ *
+ * @deprecated 9.2 see DbUtils::getEntitiesRestrictCriteria()
  **/
 function getEntitiesRestrictCriteria($table = '', $field = '', $value = '',
                                      $is_recursive = false, $complete_request = false) {
-
-   // !='0' needed because consider as empty
-   if (!$complete_request
-       && ($value != '0')
-       && empty($value)
-       && isset($_SESSION['glpishowallentities'])
-       && $_SESSION['glpishowallentities']) {
-
-      return [];
-   }
-
-   if (empty($field)) {
-      if ($table == 'glpi_entities') {
-         $field = "id";
-      } else {
-         $field = "entities_id";
-      }
-   }
-   if (!empty($table)) {
-      $field = "$table.$field";
-   }
-
-   if (!is_array($value) && strlen($value) == 0) {
-      $value = $_SESSION['glpiactiveentities'];
-   }
-
-   $crit = [$field => $value];
-
-   if ($is_recursive === 'auto' && !empty($table) && $table != 'glpi_entities') {
-      $item = getItemForItemtype(getItemTypeForTable($table));
-      if ($item !== false) {
-         $is_recursive = $item->maybeRecursive();
-      }
-   }
-
-   if ($is_recursive) {
-      $ancestors = [];
-      if (is_array($value)) {
-         foreach ($value as $val) {
-            $ancestors = array_unique(array_merge(getAncestorsOf('glpi_entities', $val),
-                  $ancestors));
-         }
-         $ancestors = array_diff($ancestors, $value);
-
-      } else if (strlen($value) == 0) {
-         $ancestors = $_SESSION['glpiparententities'];
-
-      } else {
-         $ancestors = getAncestorsOf('glpi_entities', $value);
-      }
-
-      if (count($ancestors)) {
-         if ($table == 'glpi_entities') {
-            if (!is_array($value)) {
-               $value = [$value => $value];
-            }
-            $crit = ['OR' => [$field => $value + $ancestors]];
-         } else {
-            $recur = (empty($table) ? 'is_recursive' : "$table.is_recursive");
-            $crit = ['OR' => [$field => $value,
-                              'AND' => [$recur => 1,
-                                        $field => $ancestors]]];
-         }
-      }
-   }
-   return $crit;
+   $dbu = new DbUtils();
+   return $dbu->getEntitiesRestrictCriteria(
+      $table,
+      $field,
+      $value,
+      $is_recursive,
+      $complete_request
+   );
 }
 
+/**
+ * Should APCu cache be used
+ *
+ * @return boolean
+ *
+ * @deprecated @see DbUtils::useCache()
+ */
 function useCache() {
```

Comment:
I'm not sure about this function should be here (or in DbUtils), maybe better in Toolbox or CommonGLPI

Reply:
Indeed; I'll move it to Toolbox (commondbtm seems too restrictive as well).
And since this is a very new function, I guess it should be dropped entirely from db.function.php.